src/lib/Algebra.hs

```diff
@@ -53,12 +53,13 @@ data SumClampPolynomial = SumClampPolynomial ClampPolynomial Var deriving (Show,
 elemCount :: ScalarTableType -> ClampPolynomial
 elemCount t = case t of
   BaseTy _  -> liftC $ poly [(1, mono [])]
-  TabTy v _ -> (offsets t) `psubstSumVar` (indexSetSize $ varType v)
+  TabTy b _ -> (offsets t) `psubstSumVar` (indexSetSize $ binderType b)
   _ -> error $ "Not a ScalarTableType: " ++ pprint t
 
 offsets :: ScalarTableType -> SumClampPolynomial
 offsets t = case t of
-  TabTy v body -> sumC v $ elemCount body
+  -- TODO: not sure about `fromBind` here`
+  TabTy b body -> sumC (fromBind "_" b) $ elemCount body
```

Comment:
Well yeah... We should only do `sumC` if the binder is a regular variable (as it always used to be) and otherwise we should throw a hard error for now. I think that in the future a valid thing to do would be to iterate over all variables bound in the pattern and run `sumC` for every one of them, since summing over a product type is the same as doing multiple sums over its components.